openml-python-common/src/main/java/com/feedzai/openml/python/jep/instance/JepInstance.java

```diff
@@ -62,9 +62,56 @@
     private final BlockingQueue<AbstractJepEvaluation> evaluationQueue;
 
     /**
-     * Constructor.
+     * The singleton instance that interacts with {@link jep.Jep}.
+     *
+     * @since 0.1.5
+     */
+    private static JepInstance jepInstance;
+
+    /**
+     * The lock used to initialize the singleton {@link #jepInstance}.
+     *
+     * @since 0.1.5
+     */
+    private final static Object instanceLock = new Object();
+
+    /**
+     * Gets the {@link JepInstance} singleton instance.
+     *
+     * @return The instance.
+     * @since 0.1.5
      */
-    public JepInstance() {
+    public static JepInstance getJepInstance() {
+        if (jepInstance == null) {
+            synchronized (instanceLock) {
+                if (jepInstance == null) {
+                    try{
+                        jepInstance = new JepInstance();
+                        // Start the Jep instance thread
+                        jepInstance.start();
+                    }catch(Exception e){
+                        throw new RuntimeException("Exception occurred in creating singleton jep instance");
+                    }
+                }
+            }
+        }
+        return jepInstance;
+    }
+
+    /**
+     * Private constructor for singleton pattern.
+     *
+     * @implNote {@link Jep} is initalized inside the current JVM. Due to the need to manage a consistent Python thread
```

Comment:
kudos for the explanation! IMHO I would prefer to keep the explanation generic instead of base it on the specific case we caught (and then with a reference to the specific github issue), but it's a super minor nitpick, and please let anyone else comment on this before changing anything